base/show.jl

```diff
@@ -429,6 +429,11 @@ function show_circular(io::IOContext, @nospecialize(x))
     return false
 end
 
+# An AnnotatedIOBuffer specialisation that needs to be made
+# here as `annotated.jl` is loaded earlier.
+get(io::IOContext{AnnotatedIOBuffer}, key, default) =
+    if key === :color false else get(io.dict, key, default) end
```

Comment:
```suggestion
    key === :color ? false : get(io.dict, key, default)
```

Reply:
It seems a little strange that you are dispatching on the element type here. That is type piracy, and can be unreliable. Can we do this more carefully in the constructor of `IOContext` instead (which should directly inherit the `:color` option by querying AnnnotatedIOBuffer on construction)

Reply:
Well, we want to make sure that no ANSI codes are printed to an `AnnotatedIOBuffer`. That said. since `get(::IOBuffer, :color, false) = false` maybe this isn't needed at all?

Reply:
What was the consensus here? Seems like this should be removed?

Reply:
I'm not sure this was ever discussed enough for there to be a consensus, but I think so. I'll get rid of it for now, should it turn out we want to do anything special to prevent styling control codes from being printed into an `AnnotatedIOBuffer` we can cross that bridge when we come to it.

Reply:
Styling controls are not wrong either, if the user specifically directed them to be printed into the textual part of the buffer (bypassing the annotations). That may be required to be handled anyways as well, because of the semi-frequent pattern of doing temporary buffers: `print(io, repr(thing; context=io))`. We should eventually try to ensure that usually methods that take as argument a `context isa AnnotatedIO` that those do return an `AnnotatedString` however, to avoid most of those issues.